Gems/LyShine/Code/Source/UiElementGroupComponent.cpp

```diff
@@ -0,0 +1,163 @@
+{
+    UiElementBus::Event(GetEntityId(), &UiElementInterface::SetIsRenderEnabled, enabled);
+    m_isRenderingLocallyEnabled = enabled;
+    return true;
+}
+
+/*=============*/
+/*== Utility ==*/
+/*=============*/
+
+bool UiElementGroupComponent::GetRenderingState()
+{
+    UiElementBus::EventResult(m_isRenderingLocallyEnabled, GetEntityId(), &UiElementInterface::IsRenderEnabled);
+    
+    return m_isRenderingLocallyEnabled; 
+}
+
+/*============*/
+/*== System ==*/
+/*============*/
+
+void UiElementGroupComponent::Reflect(AZ::ReflectContext* context)
+{
+    if (auto serializeContext = azrtti_cast<AZ::SerializeContext*>(context))
+    {
+        serializeContext->Class<UiElementGroupComponent, AZ::Component>()
+        ->Version(1)
+        ->Field("LocalInteraction", &UiElementGroupComponent::m_isInteractionLocallyEnabled)
+        ->Field("LocalRendering", &UiElementGroupComponent::m_isRenderingLocallyEnabled)
+        ;
+        
+        if (AZ::EditContext* editContext = serializeContext->GetEditContext())
+        {
+            editContext->Class<UiElementGroupComponent>("ElementGroup", "A grouping handler that allows interaction and rendering for the entire hierarchy of children.")
+            ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
+            ->Attribute(AZ::Edit::Attributes::Category, "UI")
+            ->Attribute(AZ::Edit::Attributes::Icon, "Icons/Components/Component_Placeholder.svg")
+            ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("UI"))
+            ->DataElement(AZ::Edit::UIHandlers::Default, &UiElementGroupComponent::m_isInteractionLocallyEnabled, "Local Interaction", "Whether this group and children will be interactable.")
+            ->DataElement(AZ::Edit::UIHandlers::Default, &UiElementGroupComponent::m_isRenderingLocallyEnabled, "Local Rendering", "Whether this group and children will be rendered.")
+            ;
+        }
+    }
+    
+    if (AZ::BehaviorContext* behaviorContext = azrtti_cast<AZ::BehaviorContext*>(context))
+    {
+        behaviorContext->EBus<UiElementGroupBus>("UiElementGroupBus")
+        ->Event("SetInteractivity", &UiElementGroupInterface::SetInteractivity)
+        ->Event("SetRendering", &UiElementGroupInterface::SetRendering)
+        ;
+    }
+}
+
+void UiElementGroupComponent::GetProvidedServices(AZ::ComponentDescriptor::DependencyArrayType& provided)
+{
+    provided.push_back(AZ_CRC_CE("UiElementGroupComponentService"));
+}
+
+void UiElementGroupComponent::GetIncompatibleServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& incompatible)
+{
+    incompatible.push_back(AZ_CRC_CE("UiElementGroupComponentService"));
+}
+
+void UiElementGroupComponent::GetRequiredServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& required)
+{
+}
+
+void UiElementGroupComponent::GetDependentServices([[maybe_unused]] AZ::ComponentDescriptor::DependencyArrayType& dependent)
+{
+}

```

Comment:
Does this really require no other services?  Just making sure, this would imply you can put one of these components on an otherwise empty entity and it will work without anything.

Thats fine if its intentional, but rare.  Normally you'd list things in here that provide the busses that it talks to, like, `UiElementBus` would imply it should have a ui element component on the same entity, 
required would them be `required.push_back(AZ_CRC_CE("UiElementService"));` or possibly dependent if its optional.

It makes hte inspector pop up with "hey, this component needs another one to work" and show the list of compatible ones that provide that service.